File: 1. Two Sum/go/two_sum.go
package two_sum

func twoSum(nums []int, target int) []int {
	prev := make(map[int]int)
	for i, v := range nums {
		res := target - v
		val, ok := prev[res]
		if ok {
			return []int{val, i}
		} else {
			prev[v] = i
		}
	}
	return []int{}
}

File: 1. Two Sum/python/two_sum.py
from typing import List


class Solution:
    def twoSumS(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i,v in enumerate(nums):
            l = target - v
            if l in d:
                return d[l], i
            else:
                d[v] = i
        return 0,0
    
    def twoSumF(self, nums: List[int], target: int) -> List[int]:
        prev = {}
        for i in range(len(nums)):
            pas = prev.get(target - nums[i])
            if not pas is None:
                return [pas, i]
            prev[nums[i]] = i
        